### PR TITLE
Add event driven architecture with DynamoDB streams post

### PR DESCRIPTION
**Pull request by alexdebrie**
- Is it too long? Should I break into two by pulling the "Challenges" section into another post?

- Need to break up some of the larger text sections with images.

### CODE REVIEW

**Comment by worldsoup** [comment]
nice! ya I think making it a two-part post with the second part focused on challenges would make a lot of sense.

File: posts/2017-06-12-event-driven-architecture-dynamodb.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+title: Event-driven processing with Serverless and DynamoDB streams
+description: Learn how to use DynamoDB streams to build event-driven architectures
+date: 2017-06-12
+layout: Post
+thumbnail: https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/variables.jpg
+authors:
+  - alexdebrie
+---
+
+In the last few years, we've seen an explosion in interest around using event streams to power interesting applications. Many companies are turning to Kafka, a persistent, distributed log developed at LinkedIn that is now a top-level Apache project. But Kafka operations aren't free -- you'll need to manage a cluster of brokers as well as a highly-available Zookeeper ensemble. Configuring, monitoring, and repairing this infrastructure can distract you from the value you want to deliver to your customers. In this post, I'll show you how you can use DynamoDB streams to decouple your application, giving you the benefits of an event stream without the maintenance burden. I'll walk though why you might want to use an event-based architecture, as well as some pitfalls to consider.
```
Comment (rupakg) [inline posts/2017-06-12-event-driven-architecture-dynamodb.md]: Typo: I'll walk you through, why ...

File: posts/2017-06-12-event-driven-architecture-dynamodb.md [hunk 2]
```diff
@@ -194,7 +194,7 @@ First, we rollback our user creation function to the original version  under the
       },
       "awsRegion": "us-west-2",
       "eventName": "MODIFY",
-      "eventSourceARN": sourcearn,
+      "eventSourceARN": "event"source:ARN",
```
Comment (rupakg) [inline posts/2017-06-12-event-driven-architecture-dynamodb.md]: still has a typo, with an extra " in ```"event"source:ARN"```

File: posts/2017-06-12-event-driven-architecture-dynamodb.md [hunk 3]
```diff
@@ -392,4 +392,4 @@ This is a simple reprocessing script that you can run locally or invoke with a r
 
 #### Conclusion
 
-In this post, we've seen how an event-driven system with DynamoDB streams and Lambda functions are a powerful pattern for reacting to what's happening in your application. We discussed some implementation details and some gotchas to watch out for when using stream-based Lambda invocations. Now it's your turn -- tell us what you build with event-driven architectures!
\ No newline at end of file
+In this post, we've seen how an event-driven system with DynamoDB streams and Lambda functions are a powerful pattern for reacting to what's happening in your application. We discussed some implementation details and some gotchas to watch out for when using stream-based Lambda invocations. Now it's your turn -- tell us what you build with event-driven architectures!
```
Comment (rupakg) [inline posts/2017-06-12-event-driven-architecture-dynamodb.md]: typo fix: "...tell us what you **built** with event....

**Pull request merged**